script.js

```diff
@@ -0,0 +1,160 @@
+// @ts-check
+
+import { composeCreateOrUpdateTextFile } from "@octokit/plugin-create-or-update-text-file";
+import prettier from "prettier";
+import YAML from 'yaml'
+
+const { parseDocument, parse, stringify, Document } = YAML
+
+const isYamlFile = (fileName) => {
+  const yamlFilePattern = /^.*\.ya?ml$/;
+  return yamlFilePattern.test(fileName);
+};
+
+/**
+ * Add cache parameter to GitHub Actions using setup-node
+ *
+ * @param {import('@octoherd/cli').Octokit} octokit
+ * @param {import('@octoherd/cli').Repository} repository
+ * @param {object} options
+ * @param {string} [options.cache] Select which package manager you want to use for caching
+ */
+export async function script(octokit, repository, { cache = "npm" }) {
+  // Global variables used throughout the code
+  const owner = repository.owner.login;
+  const repo = repository.name;
+  const defaultBranch = repository.default_branch;
+  const branchName = "add-cache-to-node-workflows";
+  const path = ".github/workflows";
+
+  // Get info on repository branches
+  const { data: branches } = await octokit.request(
+    "GET /repos/{owner}/{repo}/branches",
+    {
+      owner,
+      repo,
+      branch: defaultBranch,
+    }
+  );
+
+  // Get SHA of repository's default branch
+  const sha = branches
+    .filter((branch) => branch.name === defaultBranch)
+    .map((branch) => branch.commit.sha)[0];
+  const branchExists = branches.some((branch) => branch.name === branchName);
+
+  // Create branch if not present
```

Comment:
I think ideally you wouldn't create a branch unless any of the workflow files are updated.

Maybe you can use https://github.com/gr2m/octokit-plugin-create-pull-request, but I'm not sure if it is smart enough not to create a branch if none of the files were updated, you'd have to double check.

Reply:
I think it is controlling it here: https://github.com/gr2m/octokit-plugin-create-pull-request/blob/6ccbf00a9092fbaaf666cd0185764faa3e8738ae/src/compose-create-pull-request.ts#L117-L119

---

I'm going to give it a try. Thanks for sharing!

Reply:
It works!